File: 백준/Silver/2164. 카드2/카드2.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        Deque<Integer> q = new ArrayDeque<>();
        for (int i=1; i<=n; i++) {
            q.add(i);
        }
        while (q.size() > 1) {
            q.poll();
            if (q.size() == 1)
                break;
            q.add(q.poll());
        }
        System.out.println(q.poll());
    }
}

File: 백준/Silver/2164. 카드2/카드2.kt
import java.io.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val queue = ArrayDeque<Int>((1..n).toList())
    while (queue.size > 1) {
        queue.removeFirst()
        queue.addLast(queue.removeFirst())
    }
    println(queue.first())
}